File: Plugins/Aspose_Imaging_Java_for_Ruby/lib/asposeimagingjava/images/ditheringimage.rb
module Asposeimagingjava
  module DitheringImage
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Load an existing image
        image = Rjb::import('com.aspose.imaging.Image').load(data_dir + "test.jpg")

        # Perform Floyd Steinberg dithering on the current image
        image.dither(Rjb::import('com.aspose.imaging.DitheringMethod').ThresholdDithering, 4)

        # Save the image to disk
        image.save(data_dir + "DitheringImage.jpg")

        # Display Status.
        puts "Dithering image successfully!"
    end
  end
end

File: Plugins/Aspose_Imaging_Java_for_PHP/src/aspose/imaging/ManagingRasterFormats/DitheringImage.php
<?php
namespace Aspose\Imaging\ManagingRasterFormats;

use com\aspose\imaging\Image as Image;
use com\aspose\imaging\DitheringMethod as DitheringMethod;
class DitheringImage{

    public static function run($dataDir=null){

        # Load an existing image
        $image=new Image();
        $image = $image->load($dataDir . "test.jpg");

        # Perform Floyd Steinberg dithering on the current image
        $ditheringMethod=new DitheringMethod();
        $image->dither($ditheringMethod->ThresholdDithering, 4);

        # Save the image to disk
        $image->save($dataDir . "DitheringImage.jpg");

        # Display Status.
        print "Dithering image successfully!".PHP_EOL;
    }

}
?>